pkg/version/compatibility.go

```diff
@@ -62,3 +73,45 @@ var CompatibilityMatrix = Compatibility{
 }
 
 var DevVersionPrefix = "dev"
+
+// DeploymentVersionCompatible returns true if the given component version
+// is compatible with the installed version of Kuma CP.
+// For all binaries which share a common version (Kuma DP, CP, Zone CP...), we
+// support backwards compatibility of at most two prior minor versions.
+func DeploymentVersionCompatible(kumaVersionStr string, componentVersionStr string) bool {
+	if strings.Contains(kumaVersionStr, "dev") || strings.Contains(componentVersionStr, "dev") {
+		return true
+	}
+
+	kumaVersion, err := semver.NewVersion(kumaVersionStr)
+	if err != nil {
+		// Assume some kind of dev version
+		log.Info("cannot parse semantic version", "version", kumaVersionStr)
+		return true
+	}
+
+	componentVersion, err := semver.NewVersion(componentVersionStr)
+	if err != nil {
+		// Assume some kind of dev version
+		log.Info("cannot parse semantic version", "version", componentVersionStr)
+		return true
+	}
+
+	minMinor := int64(kumaVersion.Minor()) - 2
+	if minMinor < 0 {
+		minMinor = 0
+	}
+
+	maxMinor := kumaVersion.Minor() + 2
+
+	constraint, err := semver.NewConstraint(
+		fmt.Sprintf(">= %d.%d, <= %d.%d", kumaVersion.Major(), minMinor, kumaVersion.Major(), maxMinor),
+	)
+
+	if err != nil {
+		// Programmer error
+		panic(err)
+	}
+
+	return constraint.Check(componentVersion)
```

Comment:
Should we flag if it's the same minor but it's a patch version not equal to the CPs?

Reply:
> Should we flag if it's the same minor but it's a patch version not equal to the CPs?

Since we accept different minor versions, it would seem strange to me to make a note of different patch versions.

Rather than a compatibility between two nodes, would this be more like checking that a given node is at the most recent patch level?

Reply:
Maybe